src/Rules/Node/ValidConstantFunctionRule.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TwigCsFixer\Rules\Node;
+
+use Twig\Environment;
+use Twig\Node\Expression\ConstantExpression;
+use Twig\Node\Expression\FunctionExpression;
+use Twig\Node\Node;
+
+final class ValidConstantFunctionRule extends AbstractNodeRule
+{
+    public function enterNode(Node $node, Environment $env): Node
+    {
+        if (!$node instanceof FunctionExpression) {
+            return $node;
+        }
+
+        $functionName = $node->getAttribute('name');
+        if ('constant' !== $functionName) {
+            return $node;
+        }
+
+        $arguments = $node->getNode('arguments');
+
+        if (1 !== $arguments->count()) {
+            return $node;
+        }
+
+        if (!$arguments->getNode('0') instanceof ConstantExpression) {
+            $this->addError(
```

Comment:
```
{% set var = 'App\\SomeClass::CONSTANT_NAME' %}
{% set foo = constant(var) %}
```
doesn't work ?

We could also have `constant(some_function_generating_a_string())` no ?

Reply:
Indeed, this rule is only allow literal constants. If everything is dynamic then it cannot be verified. The rule is opt in, right? 

If it becomes an issue in the future, we can make a configuration that ignores non literals and does not produce an error. But I don't want it.

Reply:
If you have a function that returns a string why not return the value ? No need to pass to constant function

Reply:
> If you have a function that returns a string why not return the value ? No need to pass to constant function

I agree, but sometimes the `some_function_generating_a_string` method might coming from another (bad) lib an you have no choice.

> The rule is opt in, right?

Sure, but I prefer to "miss" things rather than having "false positive".
I think it's better to drop this error and early return:
- People which does weird thing with `constant` won't have an error for `constant(foo())`.
- People which use constant with string won't see the difference ; I'm pretty sure you never had this error on your codebase don't u ?

> If it becomes an issue in the future, we can make a configuration that ignores non literals and does not produce an error. But I don't want it.

If we do an option, it would be ignored by default IMHO.
But I'm not sure it's worth an option to forbid non-constant-string for the same reason, I don't think you'll have a single non-constant-string in your codebase. Am I wrong ?

Reply:
FWIW as this is already merged (thanks 😁)
I think [nobody uses `constant` function like that](https://github.com/search?q=lang%3Atwig+%22constant%28%22&type=code).

Reply:
> FWIW as this is already merged. I think [nobody uses `constant` function like that](https://github.com/search?q=lang%3Atwig+%22constant%28%22&type=code).

I agree and think the same. I just prefer to avoid the error just in case.
I bet on the fact that since nobody uses constant with something else than a string, the error isn't needed (since it will never happen on your code base).

> (thanks 😁)

Thanks you for all the good features :)